Source/Core/VideoCommon/IndexGenerator.cpp

```diff
@@ -328,13 +328,21 @@ void IndexGenerator::AddExternalIndices(const u16* indices, u32 num_indices, u32
 
 u32 IndexGenerator::GetRemainingIndices(OpcodeDecoder::Primitive primitive) const
 {
-  u32 max_index = USHRT_MAX;
+  u32 max_index = UINT16_MAX;
 
   if (g_Config.UseVSForLinePointExpand() && primitive >= OpcodeDecoder::Primitive::GX_DRAW_LINES)
     max_index >>= 2;
 
-  // -1 is reserved for primitive restart
-  max_index = max_index - 1;
+  // Although we reserve UINT16_MAX for primitive restart, we aren't allowed to use that as an
+  // actual index. But, the maximum number of vertices a game can send is UINT16_MAX, so up to
+  // 0xffff indices will be used by the game. These indices would be 0x0000 through 0xfffe,
+  // and since m_base_index gets incremented for each index used, after that m_base_index
+  // would be 0xffff and no indices remain. If a game uses 0xfffe vertices, assuming m_base_index
+  // started at 0 it would end at 0xfffe and one more index could be used. So, we do not need to
+  // subtract 1 from max_index to correctly reserve one index for primitive restart.
+  //
+  // Pocoyo Racing uses a draw command with 0xffff vertices, which previously caused issues; see
+  // https://bugs.dolphin-emu.org/issues/13136 for details.
```

Comment:
Do we have any fifologs that test the primitive restart behavior?

Reply:
Yes, both Mario Tennis fifologs; see #10414.

Reply:
Oh, though I guess I should clarify: the primitive restart referred to here is different from the one used on the GameCube/Wii. Here, primitive restart is used so that we can do a single draw call with a long triangle strip, even when the input is composed of multiple triangle strips (or even individual triangles). On the gamecube/wii, at least for mario tennis, it seems more like something where some vertices contain invalid data and need to be skipped (and there's less of a cost for doing multiple draw calls, I think?). On most platforms, dolphin uses primitive restart for *everything*, even converting lists of individual triangles into triangle strips composed of 3 vertices and then a primitive restart.

Reply:
Ok, just wanted to make sure the removal of the `-1` was tested.  I remember the primitive restart functionality, Stenzek and I had to debug my card...it turned out Vulkan+primitive restart didn't work for it :D

Reply:
Yeah, that `-1` functionality still exists (it's handled in the vertex loaders, not in IndexGenerator). IndexGenerator also still has its -1 (`s_primitive_restart`).